### ISSUE TITLE
upgrade.py --quiet is not quiet when pulling changes, effects cron

### ISSUE THREAD

**Issue by diablodale** [SpiderLabs/owasp-modsecurity-crs#718]
Using CRS v3.0.0 https://github.com/SpiderLabs/owasp-modsecurity-crs/commit/d1692bae42a0a1de81b074187250c71ded5a1d13
Ubuntu Server 16.04.2 LTS

I have cron setup with
`/etc/modsecurity/owasp-modsecurity-crs/util/upgrade.py --crs --geoip --cron --quiet && systemctl reload apache2`
Cron sent me unwanted email because upgrade.py caused git to generate progress output which is sent on stderr (as per git docs). The email received by me and sent by cron had a body of the following
```
From https://github.com/SpiderLabs/owasp-modsecurity-crs
   c3c766f..b00ae87  v3.0/dev   -> origin/v3.0/dev
```

Likely related to this line
https://github.com/SpiderLabs/owasp-modsecurity-crs/blob/v3.0/master/util/upgrade.py#L35

In the context of upgrade.py, it is not that interesting to see updates from branches other than master. In addition, this output was not captured by the subprocess.check_output() likely because this progress output occurred on stderr (as per git docs) and the code above on line 35 doesn't capture stderr using something like the following from the python docs:
```
subprocess.check_output("ls non_existent_file; exit 0", stderr=subprocess.STDOUT, shell=True)
```

The code does not successfully distinguish the following states: no-change, change, error. As written, the code blurs change and error together. It also considers changed=true when there was a commit on a dev branch. This doesn't seem to match intention. There is no change in that scenario; no change to active files in CRS because those are from master.

There are several remedies which may resolve this. What is needed is clarity on intention so the correct approach can be used. For example:
1. Route stderr to stdout when in quiet mode using `stderr=subprocess.STDOUT` to combine outputs while still supporting the handling for quiet and changed.
2. Add the `--no-progress` or `--quiet` params to the git call to silence these progress messages being sent on stderr.
3. Use exit codes from git, checking for non zero to additionally help distinguish good/bad in quiet mode
4. Alter the git setup/pull so it only operates on the master branch

Opinions?

**Comment by dune73** [comment]
Thank you for this detailed analysis. An exemplary example of a very good bug report. We'll look into improving this along your suggestions.

**Comment by victorhora** [comment]
Hey there @diablodale. Thanks for spotting this.

I just pushed [#cfc95ee](https://github.com/SpiderLabs/owasp-modsecurity-crs/commit/cfc95ee6a22ccd8c61ee19aa5c0f8f15414c7675) to fix this issue with some of your suggestions:

1. stderr is being routed to stdout with "stderr=STDOUT"
2. There's a check for a non-zero exit code to distinguish good/bad output in quiet mode
3. The git pull command is only operating on the master branch

I did some tests here and it looks good. 

One thing to keep in mind is that if you have checked out the 3.0/dev branch, the --fast-forward option will not work on 3.0/master naturally :)

Thanks.
